cl/search/management/commands/cl_index_parent_and_child_docs.py

```diff
@@ -53,21 +75,174 @@ def log_last_document_indexed(
+                    pgh_created_at__lte=end_date,
+                )
+                .order_by("id", "pgh_created_at")
+                .distinct("id")
+            )
+    return oldest_unique_events
+
+
+def get_documents_to_update_or_remove(
+    events_to_update: QuerySet,
+    search_type: str,
+    event_doc_type: EventTable | None,
+    chunk_size: int,
+) -> tuple[list[int], list[tuple[int, list[str]]], list[int]]:
+    """Determines the documents to update or remove based on changes in
+    specified fields.
+
+    :param events_to_update: A queryset of the event table instances to update.
+    :param search_type: The search type related to the update/remove action.
+    :param event_doc_type: An optional EventTable enum member specifying the
+    document type to be processed.
+    :param chunk_size: The number of items to retrieve from DB at a time.
+    :return: A four tuple containing, a list of strings containing the fields
+    to be updated, a list containing IDs of parent documents to update, a list
+    containing IDs of child documents to update, a list containing IDs of
+    documents to delete.
+    """
+
+    match search_type:
+        case SEARCH_TYPES.RECAP if event_doc_type == EventTable.DOCKET:
+            tracked_set = Docket.es_rd_field_tracker
+            fields_map = recap_document_field_mapping["save"][Docket][
+                "docket_entry__docket"
+            ]
+            document_model = Docket
+            related_child_name = "docket_entries"
+
+        case SEARCH_TYPES.RECAP if event_doc_type == EventTable.DOCKET_ENTRY:
+            tracked_set = DocketEntry.es_rd_field_tracker
+            fields_map = recap_document_field_mapping["save"][DocketEntry][
+                "docket_entry"
+            ]
+            document_model = DocketEntry
+            related_child_name = "recap_documents"
+
+        case SEARCH_TYPES.RECAP if event_doc_type == EventTable.RECAP_DOCUMENT:
+            tracked_set = RECAPDocument.es_rd_field_tracker
+            fields_map = recap_document_field_mapping["save"][RECAPDocument][
+                "self"
+            ]
+            document_model = RECAPDocument
+            related_child_name = ""
+
+        case _:
+            return [], [], []
+
+    main_documents_to_update = []
+    child_documents_to_update = []
+    documents_to_delete = []
+    event_ids = list(events_to_update.values_list("id", flat=True))
+    for event_ids_chunk in chunks(event_ids, chunk_size):
+        # Fetch event objects and current instances in bulk for the current
+        # chunk, thereby minimizing database queries and mitigating memory
+        # issues simultaneously.
+        event_ids = list(event_ids_chunk)
+        events_bulk = {
+            event.id: event
+            for event in events_to_update.filter(id__in=event_ids)
+        }
+        current_instances_bulk = document_model.objects.filter(
+            pk__in=event_ids
+        ).in_bulk()
```

Comment:
is there a reason to refine the queryset using the `filter` method instead of just passing the list of event ids to the `in_bulk` method?

Reply:
Oh yeah, the filter is not required here. It's enough to do `document_model.objects.in_bulk([ids])`, changed!